TreasureHunt/Assets/2.Scripts/Hunter/GameListView/PopupController.cs

```diff
@@ -0,0 +1,177 @@
+	{
+		// if there is already a popup, do not open another one
+		if (GameObject.FindWithTag("Popup") != null) 
+		{
+			return;
+		}
+		Debug.Log (gameID);
+		// get data
+		if (!gameObject.GetComponent<NetworkManager> ().enabled)
+		{
+			// Todo: get user id and pass it as an argument, instead of gg
+			GetContent ("gg", isMyGame);		
+		}
+		// make popup
+		GameObject popup = null;
+		if (isMyGame) 
+		{
+			popup = (GameObject)Resources.Load ("MyGamePopup");
+		} 
+		else 
+		{
+			popup = (GameObject)Resources.Load ("SearchGamePopup");
+			Debug.Log ("wow");
+		}
+		// set attributes
+		GameObject newPopup = (GameObject) Instantiate (popup);
+		newPopup.transform.SetParent (GameObject.Find ("Canvas").transform);
+		newPopup.transform.localScale = Vector3.one;
+		newPopup.transform.localPosition = new Vector3(0, 0, 0); 
+		newPopup.transform.tag = "Popup";
+		Debug.Log ("yes");
+		// find the right match for the game popup is called for
+		// parse data
+		var jsonData = JSON.Parse (data);
+		var games = jsonData ["Games"];
+		for (int i = 0; i < games.Count; i++) {
+			var curr = games [i];
+			if (String.Compare(curr ["game_id"], gameID)==0) {
+				Debug.Log ("I'm in if");
+				// attach popup attributes
+				newPopup.transform.FindChild ("Changing Objects/GameName").GetComponent<Text>().text = curr ["game_name"];
+				newPopup.transform.FindChild ("Changing Objects/Whom").GetComponent<Text>().text = curr ["maker_id"];
+				newPopup.transform.FindChild ("Changing Objects/howManyPart").GetComponent<Text>().text = curr ["participant"];
+				newPopup.transform.FindChild ("Changing Objects/howManyTrea").GetComponent<Text> ().text = curr ["treasure_count"];
+				// assign onClick event to BackButton
+				Transform backButton = newPopup.transform.FindChild("BackButton");
+				Button backB = backButton.GetComponent<Button> ();
+				backB.onClick.AddListener (TurnDownPopup);
+				// assign onClick event to DropOutButton/JoinButton
+				if(isMyGame)
+				{
+					Transform dropOutButton = newPopup.transform.FindChild("DropOutButton");
+					Button dropB = dropOutButton.GetComponent<Button> ();
+					dropB.onClick.AddListener (() => DropOrJoin(true, "gg", curr["game_id"])); // todo: put valid usn instead of "gg"
+				}
+				else
+				{
+					Transform joinButton = newPopup.transform.FindChild ("JoinButton");
+					Button joinB = joinButton.GetComponent<Button> ();
+					joinB.onClick.AddListener (() => DropOrJoin (false, "gg", curr ["game_id"])); // todo: same as above
+				}
+				// attach treasure list
+				// parse treasures
+				var treasures = curr ["Treasures"];
+				// check if there is an error
+				if (curr["treasure_count"].AsInt != treasures.Count) {
+					Debug.Log ("something wrong with treasure_count");
+				}
+				// make treasure objects 
+				for (int j = 0; j < treasures.Count; j++) 
+				{
+					// todo: 
+				}
+			}
+		}
+	}
+
+	public void TurnDownPopup(){
+		Debug.Log ("called");
```

Comment:
Ah. I believe the convention in the style guide is to configure the editor to use spaces for the tabs. But(!) it's more important that the team is consistent. Again something to agree upon with the C# code.